### PR TITLE
Updated roslyn version to consume the latest nightly, and updated cod…

### PR DESCRIPTION
**Pull request by 333fred**
…e to compile.
Tagging @dotnet/analyzer-ioperation @genlu @jinujoseph for review.

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <configuration>
   <packageSources>
     <clear />
-    <add key="myget.org roslyn-master-nightly" value="https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json" />
+    <add key="myget.org roslyn" value="https://dotnet.myget.org/F/roslyn/api/v3/index.json" />
```
Comment (jinujoseph) [inline NuGet.config]: I thought the intention was to pick from nightly drop 
@mavasani ?

Comment (333fred) [inline NuGet.config]: I think this is nightly? I can't find the original repo (roslyn-master-nightly) on myget, and the roslyn repo appears to be updated with every build. Am I missing permissions?

Comment (jinujoseph) [inline NuGet.config]: you may be right, just checked, thatz the url which was provided for dogfooding as well... not sure why manish had different url.

**Comment by mavasani** [comment]
Please don't merge this in right now. We need following pieces to happen before we can move to newer Roslyn packages:

1. Release 2.0.0-beta1 NuGet and VSIX packages for analyzers. Otherwise, we cannot take any further fixes to analyzers and fixes into the analyzer VSIX we want to release for 15.3
2. Fix all the breaking changes and important issues in Roslyn: Gen's IInvocation fixes, the IOperationWithChildren changes and any other important IOperation bug fixes.

File: NuGet.config (hunk 1 above)
Comment (mavasani) [inline NuGet.config]: I picked the url from roslyn master NuGet.config: https://github.com/dotnet/roslyn/blob/master/NuGet.Config#L24

File: build/Targets/References.Targets [hunk 2]
```diff
@@ -54,7 +54,7 @@
   <Choose>
     <When Condition="'$(IncludeCodeAnalysisReference)' == 'true'">
       <ItemGroup>
-        <PackageReference Include="Microsoft.CodeAnalysis" Version="2.0.0" />
+        <PackageReference Include="Microsoft.CodeAnalysis" Version="3.0.0-beta1-61628-02" />
```
Comment (mavasani) [inline build/Targets/References.Targets]: Interesting, why is this 3.0.0-beta1 instead 2.3.0-beta1

**Comment by mavasani** [comment]
Sorry my previous comment was not fully correct - For (1), we want to release 2.0.0-beta1 NuGet packages that can work on Dev15 (not 15.3), hence we need to wait for any fixes that need to go into it before we move Roslyn forward. Additionally, if we want to release a VSIX on gallery that works on Dev15, we can consider that, but it is probably not highest priority.

**Comment by genlu** [comment]
@mavasani For releasing dev15 compatible packages, we probably can create a 2.0 branch, all fixes should go into that branch (with daily code flow into master, which will be for 2.3). Then we can merge this PR into master now.

File: build/Targets/References.Targets (hunk 2 above)
Comment (genlu) [inline build/Targets/References.Targets]: 3.0 is not from master, it's from dev16.

**Comment by mavasani** [comment]
I still believe we should wait for all important fixes in Roslyn master to go through before moving our repo to newer Roslyn packages.

**Comment by genlu** [comment]
Sure, we can wait those important fixes for master (analyzer2.3), but any analyzer2.0 related work doesn't depend on that, actually, taking a dependency on those changes means no longer compatible with Roslyn2.0/dev15.0 because of the interface change in roslyn2.3, right?

**Comment by jinujoseph** [comment]
Fred is aware of Gen's changes and will wait for those to land before merging this in (with fixes to broken unit test that will be caused by the changes ) 
Agree with Gen, once we release 2.0.0-beta1 out of this repo, we should snap master to 2.0 branch and continue all 2.3 changes in master. 

Lets discuss on monday in person, if there is any concern.

File: NuGet.config (hunk 1 above)
Comment (333fred) [inline NuGet.config]: Confirmed with @jaredpar that `roslyn-master-nightly` is an old feed.

**Comment by 333fred** [comment]
Updated with the 2.3.0 versions of roslyn and codeanalysis. I tried updating our analyzers reference to 2.0.0-beta1, but we get hit with https://github.com/dotnet/roslyn-analyzers/issues/1154, so I left them at 1.2.0-beta2. Ping @mavasani and @jinujoseph, as we've now published the 2.0.0 versions.

**Comment by 333fred** [comment]
@dotnet-bot retest Windows Release please

**Comment by mavasani** [comment]
@333fred You also need to update the repo [Version](https://github.com/dotnet/roslyn-analyzers/blob/master/build/Targets/Analyzers.Imports.targets#L8) and [PreReleaseVersion](https://github.com/dotnet/roslyn-analyzers/blob/master/build/Targets/Analyzers.Imports.targets#L10) to 2.3.0-beta1.

**Comment by mavasani** [comment]
@333fred Can you also try to remove this ruleset entry: https://github.com/dotnet/roslyn-analyzers/blob/master/build/Rulesets/Analyzers.ruleset#L27? This analyzer should work fine on the 15.3 Roslyn bits...

**Comment by jinujoseph** [comment]
closed via https://github.com/dotnet/roslyn-analyzers/pull/1203

**Pull request closed without merging**